src/UnitOfWork.php

```diff
@@ -806,8 +808,14 @@ public function flush()
             ['memory' => memory_get_usage(true) / 1024 / 1024]
         );
 
-        if ($batchResponses = $this->getClient()->executeBatch()) {
-            $this->processResponsesFromBatch($batchResponses);
+        try {
+            if ($batchResponses = $this->getClient()->executeBatch()) {
+                $this->processResponsesFromBatch($batchResponses);
+            }
+        } catch (CtApiException $exception) {
+            if (stripos($exception->getMessage(), 'Error completing request') !== false) {
+                throw new ConnectException($exception->getMessage());
```

Comment:
Could you add the previous exception to the new ConnectException?

Reply:
Good eye!